master is not compiling

Something funny happened with the recent serialization PRs:


```
/usr/include/boost/archive/detail/oserializer.hpp:310:22:   [ skipping 23 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/boost/archive/detail/interface_oarchive.hpp:70:9:   required from
‘Archive& boost::archive::detail::interface_oarchive<Archive>::operator<<(const T&)
[with T = dealii::parallel::GridTools::internal::CellDataTransferBuffer<1, unsigned int>;
Archive = boost::archive::binary_oarchive]’
/home/drwells/Documents/Code/CPP/dealii-dev/include/deal.II/distributed/grid_tools.h:210:21:
required from ‘std::vector<char> dealii::parallel::GridTools::internal::CellDataTransferBuffer<dim, T>::pack_data() const
[with int dim = 1; T = unsigned int]’
/home/drwells/Documents/Code/CPP/dealii-dev/include/deal.II/distributed/grid_tools.h:346:28:
required from ‘void dealii::parallel::GridTools::exchange_cell_data_to_ghosts
(const MeshType&, std::function<boost::optional<T>(const typename
MeshType::active_cell_iterator&)>, std::function<void(const typename MeshType::active_cell_iterator&, const DataType&)>)
[with DataType = unsigned int; MeshType = dealii::hp::DoFHandler<1, 1>;
typename MeshType::active_cell_iterator = dealii::TriaActiveIterator<dealii::DoFCellAccessor<dealii::hp::DoFHandler<1, 1>, false> >]’
/home/drwells/Documents/Code/CPP/dealii-dev/source/hp/dof_handler.cc:935:19:
required from ‘static void dealii::internal::hp::DoFHandler::Implementation::communicate_active_fe_indices(dealii::hp::DoFHandler<dim, spacedim>&) [with int dim = 1; int spacedim = 1]’
/home/drwells/Documents/Code/CPP/dealii-dev/source/hp/dof_handler.cc:1291:85:
required from ‘void dealii::hp::DoFHandler<dim, spacedim>::distribute_dofs(const dealii::hp::FECollection<dim, spacedim>&) [with int dim = 1; int spacedim = 1]’
/home/drwells/Documents/Code/CPP/dealii-dev/build/source/hp/dof_handler.inst:12:17:   required from here
/usr/include/boost/serialization/access.hpp:116:11: error: ‘struct std::array<unsigned int, 4>’ has no member named ‘serialize’
         t.serialize(ar, file_version);
         ~~^~~~~~~~~
```

Ping @tjhei (I suspect that you know the fix)

--- a/source/dofs/dof_handler_policy.cc
+++ b/source/dofs/dof_handler_policy.cc
@@ -36,6 +36,7 @@ DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
 #include <boost/iostreams/filtering_stream.hpp>
 #include <boost/iostreams/device/back_inserter.hpp>
 #include <boost/iostreams/filter/gzip.hpp>
+#include <boost/serialization/array.hpp>
 DEAL_II_ENABLE_EXTRA_DIAGNOSTICS
 
 #include <set>

--- a/source/hp/dof_handler.cc
+++ b/source/hp/dof_handler.cc
@@ -30,6 +30,8 @@
 #include <deal.II/distributed/tria.h>
 #include <deal.II/distributed/grid_tools.h>
 
+#include <boost/serialization/array.hpp>
+
 #include <set>
 #include <algorithm>
 #include <functional>

